### PR TITLE
[d3d9] Don't clamp clears to the size of RT 0

### PR DESCRIPTION
**Pull request by K0bin**
Fixes shadows in Black Mesa.

Black Mesa clears with RT | Depth. The depth buffer is 4096x4096 and the RT is 2048x2048.
It used to work by accident because DXVK mistakenly did a fast full clear.

@DadSchoorse added a check for the DS size which broke it. https://github.com/doitsujin/dxvk/commit/a75cf2f39d3ee134406e06b6d6e2e7f54f81dc9b

### CODE REVIEW

File: src/d3d9/d3d9_device.cpp [hunk 1]
```diff
@@ -1445,12 +1445,17 @@ namespace dxvk {
...
-      bool                     fullClear,
+      uint32_t                 alignment,
       VkOffset3D               offset,
       VkExtent3D               extent,
       const Rc<DxvkImageView>& imageView,
       VkImageAspectFlags       aspectMask,
       VkClearValue             clearValue) {
+      
+      VkExtent3D imageExtent = imageView->image()->info().extent;
+      const bool fullClear = align(extent.width, alignment) == imageExtent.width
```
Comment (DadSchoorse) [inline src/d3d9/d3d9_device.cpp:1456]: I would clamp the extent before this, then we can still do fast clears if the extent is too big.

Comment (DadSchoorse) [reply src/d3d9/d3d9_device.cpp:1456]: Also you need to check if the offset is 0

Comment (K0bin) [reply src/d3d9/d3d9_device.cpp:1459]: Good call.

File: src/d3d9/d3d9_device.cpp [hunk 2]
```diff
@@ -1445,12 +1445,22 @@ namespace dxvk {
...
     auto ClearImageView = [this](
-      bool                     fullClear,
+      uint32_t                 alignment,
       VkOffset3D               offset,
       VkExtent3D               extent,
       const Rc<DxvkImageView>& imageView,
       VkImageAspectFlags       aspectMask,
       VkClearValue             clearValue) {
+      
+      VkExtent3D imageExtent = imageView->image()->info().extent;
```
Comment (DadSchoorse) [inline src/d3d9/d3d9_device.cpp:1455]: This is the wrong extent if the image view isn't mip 0

**Pull request merged**